source/extensions/filters/http/jwt_authn/verifier.h

```diff
@@ -19,7 +19,7 @@ class Verifier {
   virtual ~Verifier() {}
 
   // Verify all tokens on headers, and signal the caller with callback.
-  virtual void verify(VerifyContext& context) const PURE;
+  virtual void verify(VerifyContextSharedPtr context) const PURE;
```

Comment:
Not sure why you want to change this to (headers, callback). Wouldn't verifiers own the context at that point? I thought you didn't want any dynamic data in versifiers.

Reply:
I see.  Yes, I don't want Verifier to own dynamic data.  I just want Verifier interface to be clean, to hide all implementation detail

How about this idea:

class Verifier {

   // just a empty object to outside world
   class Context {
   };

   static ContextSharePtr  createContext(headers, callback);

   virtual verify(ContextSharePtr ctx) PURE;
};

Internally,  need to down_cast it to ContextImpl to access its data.  Or maybe access headers and callback can be at base class,  other stuff in the ContextImpl